### PR TITLE
WT-11484 Disable strict aliasing for GCC based builds

### PR DESCRIPTION
**Pull request by mickgraham-mongodb**

### CODE REVIEW

**Review by quchenhao — APPROVED** [review]
lgtm

File: cmake/configs/modes.cmake [hunk 1]
```diff
@@ -222,3 +220,6 @@ else()
     string(REPLACE "-g" "" CMAKE_CXX_FLAGS_RELWITHDEBINFO ${CMAKE_CXX_FLAGS_RELWITHDEBINFO})
 endif()
 
+if(GNU_C_COMPILER OR GNU_CXX_COMPILER)
```
Comment (ajmorton) [inline cmake/configs/modes.cmake:223]: Should we set `no-strict-aliasing` for clang as well to be consistent?

**Review by ajmorton — APPROVED** [review]
LGTM. One question, no blockers

File: cmake/configs/modes.cmake (hunk 1 above)
Comment (mickgraham-mongodb) [reply cmake/configs/modes.cmake:223]: From my checking it doesn't support it.

**Review by korteland — APPROVED** [review]
LGTM - I'm a little surprised about Clang not supporting that flag, but stranger things have happened.

**Review by marcbutler — APPROVED** [review]
lgtm

**Comment by marcbutler** [comment]
> LGTM - I'm a little surprised about Clang not supporting that flag, but stranger things have happened.

I don't understand clang does support the flag:
```
ws:tmp [0]$ /opt/mongodbtoolchain/v4/bin/clang -fno-strict-aliasing -Wall -o test test.c
ws:tmp [0]$
```

@mickgraham-mongodb ?

**Comment by mickgraham-mongodb** [comment]
> > LGTM - I'm a little surprised about Clang not supporting that flag, but stranger things have happened.
> 
> I don't understand clang does support the flag:
> 
> ```
> ws:tmp [0]$ /opt/mongodbtoolchain/v4/bin/clang -fno-strict-aliasing -Wall -o test test.c
> ws:tmp [0]$
> ```
> 
> @mickgraham-mongodb ?

Looking at "clang -help" there are options for aliasing but they don't match the behaviour from what I can determine. I suspect it accepts the request, but there is no outcome from it (as the issue we originally saw with WT-10740 does not occur with Clang).

**Comment by marcbutler** [comment]
That makes sense. It accepts the flag so it can be "a drop" in replacement
for gcc.

**Comment by mickgraham-mongodb** [comment]
Build failures are known (unrelated issues).

**Pull request merged**